### ISSUE TITLE
World Advancement Progression compatibility with Passive Powers by Smoothbrain

### ISSUE THREAD

**Issue by nbusseneau** [OrianaVenture/VentureValheim#33]
Hey, thanks for the awesome mods.

It seems there is an incompatibility between [Passive Powers](https://thunderstore.io/c/valheim/p/Smoothbrain/PassivePowers/) and [World Advancement Progression](https://thunderstore.io/c/valheim/p/VentureValheim/World_Advancement_Progression/). PP allows multiple powers to be grabbed at the same time (2 by default), and optionally allows activating one of them for a short burst of power.

With WAP's `UsePrivateKeys` + `LockGuardianPower` enabled, and provided the appropriate private player keys are registered:
- If a single power is grabbed: the activation is not blocked by WAP, as expected.
- If multiple powers are grabbed: the activation is blocked by WAP (with the usual block message). Expected: the activation is not blocked by WAP.

### Sanity check

1. Install [World Advancement Progression](https://thunderstore.io/c/valheim/p/VentureValheim/World_Advancement_Progression/).
1. In WAP config, set:
    ```
    UsePrivateKeys = true
    LockGuardianPower = true
    ```
1. Create new character and start new world.
1. In console, use:
    ```
    devcommands
    spawn TrophyEikthyr
    spawn TrophyTheElder
    ```
1. Hook trophies.
1. Try to grab both powers: this is blocked by WAP, as expected.
1. In console, use:
    ```
    setprivatekey defeated_eikthyr
    ```
1. Try to grab both powers: Eikthyr is allowed by WAP but The Elder is blocked, as expected.
1. Try to use Eikthyr power with `F`: this is allowed by WAP, as expected.
1. In console, use:
    ```
    setprivatekey defeated_gdking
    setpower GP_Eikthyr
    ```
1. Try to grab The Elder power: The Elder is now allowed by WAP, as expected.
1. Try to use The Elder power with `F`: this is allowed by WAP, as expected.

### Compatibility check

1. Prerequisite: steps 1 and 2 from sanity check above.
1. Install [Passive Powers](https://thunderstore.io/c/valheim/p/Smoothbrain/PassivePowers/).
1. In PP config, set:
    ```
    Boss powers can be activated = On
    Cooldown for boss powers (seconds) = 30
    Duration for active boss powers (seconds) = 10
    Power loss duration after boss power activation (seconds) = 20
    Shortcut for boss power 1 = F
    Shortcut for boss power 2 = F + LeftShift
    ```
1. Create new character and start new world.
1. In console, use:
    ```
    devcommands
    spawn TrophyEikthyr
    spawn TrophyTheElder
    ```
1. Try to grab both powers: this is blocked by WAP, as expected.
1. In console, use:
    ```
    setprivatekey defeated_eikthyr
    ```
1. Try to grab both powers: Eikthyr is allowed by WAP but The Elder is blocked, as expected.
1. Try to use Eikthyr power with `F`: this is allowed by WAP, as expected.
1. In console, use:
    ```
    setprivatekey defeated_gdking
    ```
1. Try to The Elder power: The Elder is now allowed by WAP, as expected.
1. Try to use both powers with `F` and `F + LeftShift`: both are blocked by WAP.
    - Expected: either activation should not be blocked by WAP.
1. Release either power: activation is now allowed by WAP again, as expected.

## Potential source of conflict

Note: I've only glanced at WAP's code and the code at https://github.com/blaxxun-boop/PassivePowers/ so this might be completely wrong, but I think PP registers multiple powers by having `m_guardianPower` in `Player` be a comma-separated string of the powers, e.g. `GP_Eikthyr,GP_TheElder` when grabbing both as in the reproduction steps above. Trying to manually change powers via the console using `setpower GP_Eikthyr,GP_TheElder` at least seems to confirm this hypothesis. WAP then fails to match this because it of course does not know the power `GP_Eikthyr,GP_TheElder` when trying to match it in `HasGuardianKey`: https://github.com/OrianaVenture/VentureValheim/blob/ab9ccea04b2f51e57ce251356db21c39eda3fefd/Progression/src/KeyLockingManager.cs#L216-L229

**Comment by OrianaVenture** [comment]
Thank you for the detailed bug report. This has been a known issue with low priority to fix. It does appear the list is stored comma separated, which should be very easy for me to add compatibility in a future update.

**Comment by nbusseneau** [comment]
Cheers, I sent a pull request for this at #34, hopefully this will be a good enough stopgap solution until you revisit this later.

**Comment by OrianaVenture** [comment]
Merged, added in version 0.2.9
